core/src/test/java/org/apache/calcite/test/SqlToRelConverterTest.java

```diff
@@ -4678,4 +4678,14 @@ void checkCorrelatedMapSubQuery(boolean expand) {
         .withConformance(SqlConformanceEnum.LENIENT)
         .ok();
   }
+
+  /**
+   * Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-5297">[CALCITE-5297]
+   * Casting dynamic variable twice throws exception</a>.
+   */
+  @Test void testDynamicParameterDoubleCast() {
+    String sql = "SELECT CAST(CAST(? AS INTEGER) AS INTEGER)";
```

Comment:
[JIRA](https://issues.apache.org/jira/browse/CALCITE-5297) is about Cast Integer, then Char (_CAST(CAST(? AS INTEGER) AS CHAR)_), test is for double CAST to INTEGER. Is it intentional ?

Reply:
but not to confuse anyone with Jira ticket. I think I will return back old test case